filters/imagefilter.go

```diff
@@ -48,25 +49,12 @@ func HandleImageResponse(ctx filters.FilterContext, f ImageFilter) {
 
 	rsp.Header.Del("Content-Length")
 
-	in := rsp.Body
-	r, w := io.Pipe()
-	rsp.Body = r
-
-	go handleImageTransform(w, in, f)
+	handleImageTransform(rsp, f)
```

Comment:
see my other comment for the handleImageTransform function.

First make a `defer rsp.Body.Close()`.

You can then pass in to it the rsp.Body to handleImageTransform, and not the whole rsp. It could return a reader, that you can wrap here with the NopCloser, and replace rsp.Body with it. The original rsp.Body will be closed once the HandleImageResponse returns.